cache/ringbuffer.go

```diff
@@ -0,0 +1,93 @@
+package cache
+
+import (
+	"fmt"
+	"sync"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+type ringBuffer struct {
+	mu         sync.RWMutex
+	buffer     []*clientv3.Event
+	head, tail int
```

Comment:
Please document that head doesn't point to "head" but the index after last non-empty entry.